packages/customer-erasure/src/cli.js

```diff
@@ -0,0 +1,148 @@
+import fs from 'fs'
+import { getCredentials } from '@commercetools/get-credentials'
+import pino from 'pino'
+import PrettyError from 'pretty-error'
+import yargs from 'yargs'
+
+import CustomerErasure from './main'
+import { description } from '../package.json'
+
+process.title = 'customer-erasure'
+
+const args = yargs
+  .usage(
+    `
+Usage: $0 [options]
+${description}`
+  )
+  .showHelpOnFail(false)
+  .help('help', 'Show help text.')
+  .version()
+  .option('output', {
+    alias: 'o',
+    default: 'stdout',
+    describe: 'Path to output file.',
+  })
+  .option('customerId', {
+    alias: 'c',
+    describe: 'Customer to fetch or delete.',
+    demand: true,
+  })
+  .option('apiUrl', {
+    default: 'https://api.sphere.io',
+    describe: 'The host URL of the HTTP API service.',
+  })
+  .option('authUrl', {
+    default: 'https://auth.sphere.io',
+    describe: 'The host URL of the OAuth API service.',
+  })
+  .option('accessToken', {
+    describe: 'CTP client access token.',
+  })
+  .option('projectKey', {
+    alias: 'p',
+    describe: 'API project key.',
+    demand: true,
+  })
+  .option('deleteAll', {
+    type: 'boolean',
+    alias: 'D',
+    describe: 'Delete all data related to customer.',
+  })
+  .option('logLevel', {
+    default: 'info',
+    describe: 'Logging level: error, warn, info or debug.',
+  })
+  .option('prettyLogs ', {
+    describe: 'Pretty print logs to the terminal',
+    type: 'boolean',
+  })
+  .option('logFile', {
+    default: 'customer-erasure.log',
+    describe: 'Path to where to save logs file.',
+    type: 'string',
+  }).argv
+
+// instantiate logger
+const loggerConfig = {
+  level: args.logLevel,
+  prettyPrint: args.prettyLogs,
+}
+const logger = pino(loggerConfig)
+
+// print errors to stderr if we use stdout for data output
+// if we save data to output file errors are already logged by pino
+const logError = error => {
+  const errorFormatter = new PrettyError()
+
+  if (args.logLevel === 'debug')
+    process.stderr.write(`ERR: ${errorFormatter.render(error)}`)
+  else process.stderr.write(`ERR: ${error.message || error}`)
+}
+
+const errorHandler = errors => {
+  if (Array.isArray(errors)) errors.forEach(logError)
+  else logError(errors)
+
+  process.exitCode = 1
+}
```

Comment:
General question: what happens if there is an error in the middle of the process? For example, if I get an network error and I retry the command, would it still complete the "deletion" or would there be problems because of the interrupted process?

Reply:
That should work just fine as we let `404`s resolve and only call delete if data exists [here](https://github.com/commercetools/nodejs/pull/601/files#diff-39c284b79054f260b4baecab4f766173R179)

Reply:
I'm just asking to avoid having dead references.

And also to make sure that the script is idempotent, meaning that it can be executed multiple times.